integration/test/app/kubeconfig/kubeconfig_test.go

```diff
@@ -56,30 +67,41 @@ func TestAppLifecycleUsingKubeconfig(t *testing.T) {
 			Version: "1.0.0",
 		},
 		Namespace: namespace,
+		ConfigMap: chartvalues.APIExtensionsAppE2EConfigConfigMap{
+			ValuesYAML: `test:
+      image:
+        registry: quay.io
+        repository: giantswarm/alpine-testing
+        tag: 0.1.1`,
+		},
+		Secret: chartvalues.APIExtensionsAppE2EConfigSecret{
+			ValuesYAML: `secret: "test"`,
+		},
 	}
 
-	// TODO: Remove kubeconfig setting after implement KIND test on app-operator
-	// issue: https://github.com/giantswarm/giantswarm/issues/4606
+	// Transform kubeconfig file to restconfig and flatten
+	var bytes []byte
 	{
-		config.Logger.LogCtx(ctx, "level", "debug", "message", "creating kubeconfig secret")
+		c := clientcmd.GetConfigFromFileOrDie(env.KubeConfigPath())
 
-		// Transform kubeconfig file to restconfig and flatten
-		var bytes []byte
-		{
-			c := clientcmd.GetConfigFromFileOrDie("/workdir/.shipyard/config")
+		err = api.FlattenConfig(c)
+		if err != nil {
+			t.Fatalf("expected nil got %#v", err)
+		}
 
-			err = api.FlattenConfig(c)
-			if err != nil {
-				t.Fatalf("expected nil got %#v", err)
-			}
+		// Normally KIND assign 127.0.0.1 as server address, that should change into kubernetes
+		c.Clusters["kind"].Server = "https://kubernetes.default.svc.cluster.local"
```

Comment:
This is for workaround, no need to put TODO in here in my opinion.

Reply:
Agreed. We may need to change this for the 2nd kind cluster. But this is fine for now.